src/auth/ha-auth-form.ts

```diff
@@ -0,0 +1,46 @@
+import { CSSResultGroup } from "lit";
+import { customElement } from "lit/decorators";
+import { fireEvent } from "../common/dom/fire_event";
+import { HaForm } from "../components/ha-form/ha-form";
+import { HaFormElement, HaFormSchema } from "../components/ha-form/types";
+import "./ha-auth-form-string";
+
+@customElement("ha-auth-form")
+export class HaAuthForm extends HaForm {
+  protected fieldElementName(type: string): string {
+    if (type === "string") {
+      return "ha-auth-form-string";
+    }
+    return super.fieldElementName(type);
+  }
```

Comment:
This should be handled in core eventually, it can provide a different type for the auth text fields.

Reply:
Yes, so we don't have to duplicate ha-form but only add an option to render without shadow dom.